### ISSUE TITLE
Add support for "in context checkout" flavor of Express Checkout

### ISSUE THREAD

**Issue by pdbreen** [thephpleague/omnipay-paypal#77]
Paypal has added an in-context variation for Express checkout
https://developer.paypal.com/webapps/developer/docs/classic/express-checkout/in-context/integration/

On the server, its essentially identical - but makes use of a different redirect url which eliminates the cmd parameter and replaces `/cgi-bin/webscr` with `/checkoutnow`

Easy to work around, but would be nice to have baked into either a different gateway (PayPal_ExpressInContext?) or an optional override on the response object.

For now, munging the redirect url at arms length is enough to get it working:

```php
$gateway = Omnipay::create('PayPal_Express');

$gateway->setUsername(env('PAYPAL_USERNAME'));
$gateway->setPassword(env('PAYPAL_PASSWORD'));
$gateway->setSignature(env('PAYPAL_SIGNATURE'));
$gateway->setTestMode(env('PAYPAL_ENVIRONMENT') != 'production');

$response = $gateway->purchase(
	array(
		'cancelUrl'=> url($location),
		'returnUrl'=> url('/paypal/success'),
		'amount' =>  $order->finalOrderTotal(),
		'currency' => 'USD'
	)
)->send();

if ($response->isRedirect()) {
	$url = $response->getRedirectUrl();
	$url = preg_replace('/cgi-bin\/webscr/', 'checkoutnow', $url);
	$url = preg_replace('/cmd=_express-checkout&/', '', $url);
	return redirect($url, 302, [], true);
}
```

**Comment by delatbabel** [comment]
A PR for this would be welcome.

**Comment by vercotux** [comment]
Any news on this?

**Comment by delatbabel** [comment]
Nobody has submitted a PR for this yet, in case that's what you're asking.

**Comment by flo-sch** [comment]
Just proposed a new PR that should fix the issue, using a new `PayPal Express In-Context` gateway.

**Comment by delatbabel** [comment]
PR has been merged, just waiting on production level testing with that PR and 2 others before tagging.
